### PR TITLE
Re-Wrap recipes (Second Pass)

### PR DESCRIPTION
**Pull request by Soup-de-Loop**
#### Summary
```SUMMARY: Infrastructure "Re-wraps all recipes to follow modern standards"```

#### Purpose of change

Recipes were largely inconsistent which ill begets all of the hard work the developers have done to have jsonized so much, and provided `requirements` as a strong tool to help keep things consistent and organized.

#### Describe the solution

This is my second pass attempting this. I have manually gone through every json file within the `\cdda\data\json\recipes` including sub-folders and purposefully re-written and nudged lines to fit a neat standard, drawing from the majority entries that have been modernized as a key. 

Additionally, new requirements have been added that help establish a huge consistency problem across recipes, and many gunsmithing related recipes have been reworked to only need a soldering iron equivelant and a repair kit on hand. More difficult and heavy jobs still use a welder where appropriate, and some require the large kit as opposed to the small one. Between `gunsmithing_standard`, `plastic_making` and `crucible_standard`, many lines and characters have been reduced while still providing full functionality, and should hopefully aid future additions.

__NPC recipes including the entire basecamp folder have not been touched.__

#### Describe alternatives you've considered

Aside from not helping in general, I did some research on gunsmithing and plastic welding in order to help inform the changes to gunsmithing for gun related accessories and case-by-case consideration of plastics being needed to be melted and shaped. However, it would've been possible to leave the recipes still using welders as they do currently, and still reduce the imprint. I _can_ still apply that change instead of the more in-depth changes I've performed, should the new form be unpopular.

#### Testing

This time all changes were performed using ATOM as an extension to Github Dekstop with full markdown capabilities, aiding monumentally in catching issues. Every json that was changed was then transferred from the development branch to my current game, loaded, and tested for errors. I repeated this until there were no errors thrown in-game- Checking the recipes were working as intended was also performed ingame through the `&` menu. Before opening this PR, each file was linted, but I'm sure I'll still have to do some astyle edits once travis has had a look.

#### Additional context

TL;DR: Re-wrapping done using already existing requirements conventions; re-sorted inconsistent entries to follow same order; new requirements added to further reduce the overhead on recipes and fully integrated.

Example of new gunsmithing tools for guns more complicated than simple makeshift pipe rifles:
example
Potential issue that has arisen when multiple tool groups with many choices stack up:
example2

### CODE REVIEW

**Comment by KorGgenT** [comment]
You've got some crafting test failures here.

**Comment by Soup-de-Loop** [comment]
> You've got some crafting test failures here.

Aye, I can’t quite put my finger on it since it all seems to work ingame for my local game directory. Does the test subject perhaps have limited charges to draw from? The plastic_making requirement ends up using 5u of surface heat per plastic chunk to melt the plastic to shape- in a lot of recipes, this remains fairly consistent, but it does mean an overall increase in energy usage particularly as more plastic is used. I might try de-committing the electronic recipe changes after a minor commit to make sure it’s failing in the same place, and see if it snags elsewhere

**Comment by ZhilkinSerg** [comment]
If you've changed recipes you probably need to update tests that include these recipes. The file is `\tests\crafting_test.cpp`.

**Comment by Soup-de-Loop** [comment]
It is making a mockery of me now. Where on earth are those numbers coming from? The hotplate requirement used to be a flat 10, now it's calculated to be 30 (5u per plastic, at 6 plastic chunks for this recipe). But the test is saying that only 16 charges were drawn, leaving 24 instead of 10. Then in the next bit of the test, it adds 15 and then 15 again as I specified, but then it goes and pulls out 20 instead of 30 again?

**Comment by Soup-de-Loop** [comment]
I feel like the error being thrown by the tests is an issue with the crafting test, but I honestly don't know enough to figure it out myself. I've tried changing values and applying a few patches, and I can get it to behave normally in-game but I'm stumped in terms of the test throwing random results, and even the consistent one that's out of line is way out of line.

**Comment by kevingranade** [comment]
Please, split things like this up into smaller PRs.  If you are having any problems at all with your PR getting cleaned up for merge, shrink it.  I'm going to go ahead and close this because I suspect that it's going to be easier to start over at this point than to handle the conflicts.
